indico/modules/rb_new/client/js/common/rooms/selectors.js

```diff
@@ -15,15 +15,68 @@
  * along with Indico; if not, see <http://www.gnu.org/licenses/>.
  */
 
+import _ from 'lodash';
 import {createSelector} from 'reselect';
 
 import {RequestState} from 'indico/utils/redux';
 
 
 export const hasLoadedEquipmentTypes = ({rooms}) => rooms.requests.equipmentTypes.state === RequestState.SUCCESS;
-export const getEquipmentTypes = ({rooms}) => rooms.equipmentTypes;
+const getEquipmentTypes = ({rooms}) => rooms.equipmentTypes;
+export const getEquipmentTypeNames = createSelector(
+    getEquipmentTypes,
+    equipmentTypes => equipmentTypes.map(x => x.name)
+);
+export const getFeatures = createSelector(
+    getEquipmentTypes,
+    equipmentTypes => {
+        const features = {};
+        equipmentTypes
+            .map(eq => eq.features)
+            .forEach(eqFeatures => {
+                eqFeatures.forEach(feature => {
+                    features[feature.id] = feature;
+                });
+            });
+        return _.sortBy(Object.values(features), 'title');
+    }
+);
+export const getFeaturesForFilterButtons = createSelector(
+    getFeatures,
+    features => features.filter(f => f.showFilterButton).map(f => _.pick(f, ['icon', 'name', 'title']))
+);
+
+export const getAllRooms = createSelector(
+    ({rooms}) => rooms.rooms,
+    getEquipmentTypes,
+    (rawRooms, equipmentTypes) => {
+        equipmentTypes = equipmentTypes.reduce((obj, eq) => ({...obj, [eq.id]: eq}), {});
+        return _.fromPairs(rawRooms.map(room => {
+            const {available_equipment: equipment, ...roomData} = room;
```

Comment:
Shouldn't we somewhere use `camelizeKeys` to change `available_equipment` to `availableEquipment`?

Reply:
Something for a separate commit since it'd just add lots of clutter to this PR.